### DISCUSSION TITLE
Installation shows an empty page in Nginx?

### DISCUSSION THREAD

**Discussion by teddibaer** [datenstrom/community#640] [Ask a question] [answered]
I tried to install the current master with a current debian/nginx (with php-fpm unix domain socket) /php7.3, but all I get in the browser is an empty page (http success code, no error).

I can see in the logs that the yellow.php is started, but apart from that i cannot see any error in any of the logs (nginx, php). I enabled error debug logging in yellow but no output is generated in the yellow.log.

I can use yellow.php from the command line successfully (after adding the CoreServerULR in the system ini) and build the static website.

Any clues welcome ...

**Comment by markseuffert** [comment]
I can't reproduce this on my server. I made a brand new installation on Ubuntu Linux with Apache and it worked as expected. Can you show your `system/extensions/yellow.log`? Can you try a different web server? Do you have additional information for us?

**Reply by teddibaer** [reply]
I did some other small tests, (will do some more testing later) but:

The core.php and install.php are clearly executed as I can see the errors on the web page in case requirements are not met. Once I make the needed changes the page is empty again.

It seems to me, that just the actuall install page elements are not shown and therefore I can never complete the install and yellow gets never past that step.
(And with empty page I really mean an empty head and an empty body tag)

Update: actually the browser debug tools say that there is no data at all returned (the empty head/body is generated by the browser).

**Reply by markseuffert** [reply]
Could be a crash in PHP-FPM. Can you narrow down how far the `install.php` is executed?

**Reply by teddibaer** [reply]
It is really frustrating when something does not work and you cannot find any traces in any log files pointing you at the problem ...

However - more by accident - i found it: in my vanilla nginx install exist 2 fastcgi param files: fastcgi.conf and fastcgi_params which are nearly identical. Yellow works with one (fastcgi.conf), but not the other. The only difference is:

$ diff fastcgi.conf fastcgi_params
2d1
< fastcgi_param  SCRIPT_FILENAME    $document_root$fastcgi_script_name;

Maybe some php/fpm expert can make something out of this.
Without that line php scripting seems not to work at all as even simple test php scripts did not execute. I am not sure why I saw the yellow requirements errors in the page at some point but I did different changes in different places and I cannot really reproduce what setup I had at this point.

Anynow - regards. 
Will now (finally) start playing with yellow :-)

**Reply by markseuffert** [reply]
Have some fun, you earned it. 😁

**Comment by markseuffert** [comment, marked as answer]
This is a known problem and it really surprised me. The [Nginx documentation](https://www.nginx.com/resources/wiki/start/topics/examples/phpfcgi/) says the following: If you see a blank page in browser, please check if `SCRIPT_FILENAME` parameter is set. If this parameter is not set, PHP FPM responses 200 OK with empty content, and there is no error or warning.

As a quick fix, replace `include fastcgi_params;` with `include fastcgi.conf;` in your `nginx.conf`.

There's an example Nginx configuration at https://datenstrom.se/yellow/help/troubleshooting.
